Sources/Controllers/Map/OAMapViewTrackingUtilities.mm

```diff
@@ -186,6 +186,13 @@ - (void) onMapModeChanged
                     
 
                     _mapView.mapAnimator->resume();
+
+                    OAMapViewController *mapViewController = [OARootViewController instance].mapPanel.mapViewController;
+                    if (mapViewController.needToSwitchTo3DMode && ![self isIn3dMode])
+                    {
+                        mapViewController.needToSwitchTo3DMode = NO;
+                        [self switchMap3dMode];
```

Comment:
Seems that 3D mode is turned on regardless of whether it was turned on before or not. Looks wrong.